packages/x-charts-pro/src/internals/plugins/useChartProZoom/useChartProZoom.ts

```diff
@@ -91,43 +92,62 @@ export const useChartProZoom: ChartPlugin<UseChartProZoomSignature> = ({
     };
   }, [store, paramsZoomData]);
 
-  // Add instance methods
-  const setIsInteracting = React.useCallback(
-    (isInteracting: boolean) => {
-      store.update((prev) => ({ ...prev, zoom: { ...prev.zoom, isInteracting } }));
-    },
+  // This is debounced. We want to run it only once after the interaction ends.
+  const removeIsInteracting = React.useMemo(
+    () =>
+      debounce(() =>
+        store.update((prevState) => {
+          return {
+            ...prevState,
+            zoom: {
+              ...prevState.zoom,
+              isInteracting: false,
+            },
+          };
+        }),
+      ),
     [store],
   );
 
-  const setZoomDataCallback = React.useCallback(
-    (zoomData: ZoomData[] | ((prev: ZoomData[]) => ZoomData[])) => {
-      store.update((prevState) => {
-        const newZoomData =
-          typeof zoomData === 'function' ? zoomData([...prevState.zoom.zoomData]) : zoomData;
-        onZoomChange?.(newZoomData);
-
-        if (prevState.zoom.isControlled) {
-          return prevState;
-        }
+  // This is throttled. We want to run it at most once per frame.
```

Comment:
> We want to run it at most once per frame.

If we want this, shouldn't we be using `requestAnimationFrame`? `throttle` uses `setTimeout`, doesn't that mean that it can run more frequently than once per frame?

Reply:
Leftover from initial idea. The PR description has the reasoning